test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -500,6 +505,68 @@ public async Task Restore_LegacyPackagesConfig_WithNuGetLockFileLockedMode()
             }
         }
 
+        [Fact]
+        public async Task RestorePackagesConfig_WithExistingLockFile_LockedMode_Succeeds()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var net461 = NuGetFramework.Parse("net461");
+
+                var projectA = SimpleTestProjectContext.CreateLegacyPackageReference(
+                    "a",
+                    pathContext.SolutionRoot,
+                    net461);
+
+                var packageX = new SimpleTestPackageContext()
+                {
+                    Id = "x",
+                    Version = "1.0.0"
+                };
+                packageX.Files.Clear();
+                packageX.AddFile("lib/net461/x.dll");
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+                Util.CreateFile(Path.GetDirectoryName(projectA.ProjectPath), "packages.config",
+@"<packages>
+  <package id=""x"" version=""1.0.0"" targetFramework=""net461"" />
+</packages>");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    PackageSaveMode.Defaultv3,
+                    packageX);
+
+                // Preconditions, regular restore
+                var result = RunRestore(pathContext, _successExitCode);
+                result.Success.Should().BeTrue(because: result.AllOutput);
+                new FileInfo(projectA.NuGetLockFileOutputPath).Exists.Should().BeFalse();
+
+                // Write expected lock file
+                var templateLockFile = GetResource("NuGet.CommandLine.FuncTest.compiler.resources.pc.packages.lock.json");
```

Comment:
If it is a template, I would add a suggestion in the filename.
```suggestion
                var templateLockFile = GetResource("NuGet.CommandLine.FuncTest.compiler.resources.pc.packages.lock.template.json");
```

Reply:
it's not so much a template as there is a `search and replace` happening due to the sha value. I think my variable name is the one that's wrong rather than the file name.